cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/BasicWorkflowToken.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.workflow;
+
+import co.cask.cdap.api.workflow.WorkflowToken;
+
+import java.util.HashMap;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Implementation of the {@link WorkflowToken} interface.
+ */
+public class BasicWorkflowToken implements WorkflowToken {
+  private Map<String, Map<String, Long>> mapReduceCounters;
+
+  @Nullable
```

Comment:
I would prefer it to return null if it is not set. The reason is that in the future we want to allow setting and retrieving arbitrary properties, not just M/R counters. If a property does not exist, it will return null. It should be the same for the M/R counters (they will be serialized into a property, too).